src/Link0/ProfilerBundle/EventListener/ProfilingEventListener.php

```diff
@@ -8,88 +8,65 @@
 
 use Link0\Profiler\PersistenceHandler;
-use Link0\Profiler\Profiler;
-use Symfony\Component\Console\Event\ConsoleTerminateEvent;
-use Symfony\Component\HttpKernel\Event\GetResponseEvent;
-use Symfony\Component\Console\Event\ConsoleCommandEvent;
-use Symfony\Component\HttpKernel\TerminableInterface;
+use Link0\ProfilerBundle\Service\ProfilingService;
 
 /**
  * Class ProfilingEventListener
  *
  * Hooks into the symfony event system, and starts and stops the profiler on certain events
  *
- * Start:
- *  - console.command
- *  - kernel.request
- * Stop:
- *  - console.terminate
- *  - kernel.terminate
+ * You can configure this event listener in your services.yml in the following format
+ *
+ * services:
+ *   kernel.listener.link0profilerlistener:
+ *     class: Link0\ProfilerBundle\EventListener\ProfilingEventListener
+ *     param: [@profiling_service]
+ *     tags:
+ *       - { name: kernel.event_listener, event: console.command,   method: start }
+ *       - { name: kernel.event_listener, event: kernel.request,    method: start }
+ *       - { name: kernel.event_listener, event: console.terminate, method: stop  }
+ *       - { name: kernel.event_listener, event: kernel.terminate,  method: stop  }
  *
  * @package Link0\ProfilerBundle\EventListener
  */
-class ProfilingEventListener
+final class ProfilingEventListener
 {
     /**
      * @var \Link0\Profiler\Profiler
      */
-    private $profiler;
-
-    /**
-     * A persistence handler should be fed, since null will disable persisting profiles
-     *
-     * @param PersistenceHandler $handler
-     */
-    public function __construct(PersistenceHandler $handler = null)
-    {
-        $this->profiler = new Profiler($handler);
-    }
-
-    /**
-     * @return Profiler
-     */
-    public function getProfiler()
-    {
-        return $this->profiler;
-    }
+    private $profilingService;
 
     /**
-     * @event console.command
-     * @param ConsoleCommandEvent $event
+     * @param ProfilingService $profilingService
      */
-    public function onConsoleCommand(ConsoleCommandEvent $event)
+    public function __construct(ProfilingService $profilingService)
     {
-        $this->getProfiler()->start();
+        $this->profilingService = $profilingService;
     }
 
     /**
-     * @event kernel.request
-     * @param GetResponseEvent $event
+     * @return ProfilingService
      */
-    public function onKernelRequest(GetResponseEvent $event)
+    public function getProfilingService()
     {
-        if (!$event->isMasterRequest()) {
-            return;
-        }
-
-        $this->getProfiler()->start();
+        return $this->profilingService;
     }
 
     /**
-     * @event console.terminate
-     * @param ConsoleTerminateEvent $event
+     * Starts the profiler
      */
-    public function onConsoleTerminate(ConsoleTerminateEvent $event)
+    public function start()
     {
-        $this->getProfiler()->stop();
+        $this->getProfilingService()->start();
```

Comment:
When working with internal variables, I use `$this->profilingService` in most of the cases, and not the getter, but this could be just my personal preference.